internal/ir/ir.go

```diff
@@ -352,6 +352,20 @@ type HashMap struct {
 	Value Value
 }
 
+// SubscriptExpr is an IR Value that represents the access of an element of an array.
```

Comment:
This is not necessary for just arrays, we can use for any object that can be subscripted, so I think that a better doc for this would be:
`// Lookup instruction yields element from Index of Object`